sdk/search/Azure.Search/tests/Utilities/SearchResources.Data.cs

```diff
@@ -281,87 +284,236 @@ public partial class SearchResources
+            Tags.SequenceEqualsNullSafe(other.Tags) &&
+            ParkingIncluded == other.ParkingIncluded &&
+            SmokingAllowed == other.SmokingAllowed &&
+            LastRenovationDate.EqualsDateTimeOffset(other.LastRenovationDate) &&
+            Rating == other.Rating &&
+            // TODO: XXXXX - Unify on an Azure.Core spatial type
+            // Location.EqualsNullSafe(other.Location) &&
+            Address.EqualsNullSafe(other.Address) &&
+            Rooms.SequenceEqualsNullSafe(other.Rooms);
+
+        public override int GetHashCode() => HotelId?.GetHashCode() ?? 0;
+
+        public override string ToString()
+        {
+            string FormatRoom(HotelRoom room) => $"{{ {room} }}";
+            return
+                $"ID: {HotelId}; Name: {HotelName}; Description: {Description}; " +
+                $"Description (French): {DescriptionFr}; Category: {Category}; " +
+                $"Tags: {Tags?.ToCommaSeparatedString() ?? "null"}; Parking: {ParkingIncluded}; " +
+                $"Smoking: {SmokingAllowed}; LastRenovationDate: {LastRenovationDate}; Rating: {Rating}; " +
+                $"Location: [{Location?.Longitude ?? 0}, {Location?.Latitude ?? 0}]; " +
+                $"Address: {{ {Address} }}; Rooms: [{string.Join("; ", Rooms?.Select(FormatRoom) ?? new string[0])}]";
+        }
+
+        public SearchDocument AsDocument() =>
+            new SearchDocument()
+            {
+                ["hotelId"] = HotelId,
+                ["hotelName"] = HotelName,
+                ["description"] = Description,
+                ["descriptionFr"] = DescriptionFr,
+                ["category"] = Category,
+                ["tags"] = Tags ?? new object[0],   // OData always gives [] instead of null for collections.
+                ["parkingIncluded"] = ParkingIncluded,
+                ["smokingAllowed"] = SmokingAllowed,
+                ["lastRenovationDate"] = LastRenovationDate,
+                ["rating"] = Rating,
+                // TODO: XXXXX - Unify on an Azure.Core spatial type
+                // ["location"] = Location,
+                ["location"] = Location == null ? null : new SearchDocument()
+                {
+                    ["type"] = "Point",
+                    ["coordinates"] = new double[] { Location.Longitude, Location.Latitude },
+                    ["crs"] = new SearchDocument()
+                    {
+                        ["type"] = "name",
+                        ["properties"] = new SearchDocument()
+                        {
+                            ["name"] = "EPSG:4326"
+                        }
+                    }
+                },
+                ["address"] = Address?.AsDocument(),
+                // With no elements to infer the type during deserialization, we must assume object[].
+                ["rooms"] = Rooms?.Select(r => r.AsDocument())?.ToArray() ?? new object[0]
+            };
     }
 
     [SerializePropertyNamesAsCamelCase]
```

Comment:
This attribute still exists? Can we make it work in Track 2 with `System.Text.Json`?

Reply:
No - that's using the Track 1 library in the tests do our index creation.  We will definitely be coming up with some answer here in the near future that allows using naming policies.